### ISSUE TITLE
[asl] Type checking problem with constants

### ISSUE THREAD

**Issue by HadrienRenaud** [herd/herdtools7#961]
The following test should be allowed:
```
func test ()
begin
  constant K: integer = 5;
  var bv: bits (K) = Zeros(K);
end
```

**Why?**
Even if `K` is an unbounded integer, its uses should be equivalent as using `5` directly, so it's inlining should happen before the unbounded check.

**Plan**
Use future work on side-effects to simply ignore accesses to constants.
